Lib/pdb.py

```diff
@@ -597,11 +597,20 @@ def precmd(self, line):
         args = line.split()
         while args[0] in self.aliases:
             line = self.aliases[args[0]]
-            ii = 1
-            for tmpArg in args[1:]:
-                line = line.replace("%" + str(ii),
-                                      tmpArg)
-                ii += 1
+            for idx in range(1, 10):
```

Comment:
Do you need to check that there are at most 10 args somewhere?

Would it not be better to iterate over the args as before and parse them?

Reply:
> Do you need to check that there are at most 10 args somewhere?

args or replaceable parameters? This logic already checks for number of args when there's no `%*`. There can be more than 10 args if `%*` exists.

If you meant replaceable parameters (like `%1`), we simply can't, because `%10` is just legit - it's `%1` + `0`.

Iterating over the args can't find the replaceable parameters that do not get a match (for example, you have 2 args but you have a `%3`). You'll need to iterate over the parameters again anyway.

Reply:
> You'll need to iterate over the parameters again anyway.

Don't you just need to know how many there are?

Reply:
> Don't you just need to know how many there are?

How can I know how many there are? With `re`? The easiest way is probably iterating through them.

Also, I need more than the count. For example, `alias repeat_second "%* %2"` is a valid alias (even though it's a bit weird), but `repeat_second 1` should not work - we should catch that there's no match for `%2`.

Reply:
So `%10` doesn't work now?

Reply:
It does not work now, it did not work before, we just did not mention that it did not work before. It never worked.